packages/core/src/mutants/incremental-differ.ts

```diff
@@ -230,7 +230,12 @@ export class IncrementalDiffer {
       Object.entries(testFiles ?? {}).forEach(([fileName, oldTestFile]) => {
         const relativeFileName = toRelativeNormalizedFileName(fileName);
         const currentFileSource = currentRelativeFiles.get(relativeFileName);
-        if (currentFileSource !== undefined && oldTestFile.source !== undefined) {
+        if (currentFileSource === undefined && fileName !== '') {
+          // An empty file name means the test runner cannot report test file locations.
+          // If the current file is undefined while the test runner can report test file locations, it means it has been deleted
+          log.debug('Test file removed: %s', relativeFileName);
+          testStatisticsCollector.count(relativeFileName, 'removed', oldTestFile.tests.length);
+        } else if (currentFileSource !== undefined && oldTestFile.source !== undefined) {
```

Comment:
I'm missing new unit tests for these:

![image](https://user-images.githubusercontent.com/1828233/234116940-40a2f886-2927-44d3-836c-9dd925420eee.png)